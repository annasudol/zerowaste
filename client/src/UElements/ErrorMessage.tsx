import * as React from "react";


interface ErrorMessageProps {
    message?: string
    validationMessage?: string

}
export const ErrorMessage: React.FunctionComponent<ErrorMessageProps> = ({ message, validationMessage }): React.ReactElement => {
    if (validationMessage) {
        return (
            <p className="text-coral text-xs mt-2">{validationMessage}</p>
        )
    }
    return (
        <div className="flex flex justify-center items-center h-screen">
            <h2 className="font-gotham coral text-5xl text-center pb-4">
                {message || 'Ups, error, something went wrong'}
            </h2>
        </div>
    )
}