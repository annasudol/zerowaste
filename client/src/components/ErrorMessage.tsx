import * as React from "react";

interface ErrorMessageProps {
    message?: string;
}
export const ErrorMessage: React.FunctionComponent<ErrorMessageProps> = ({ message }): React.ReactElement => (
    <div className="flex flex justify-center items-center h-screen">
        <h2 className="font-albaSuper text-orange-500 text-5xl text-center pb-4">
            {message || 'Ups, error, something went wrong'}
        </h2>
    </div>
);
