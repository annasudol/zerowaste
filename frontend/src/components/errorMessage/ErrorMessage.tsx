import React, { FC, ReactElement } from 'react';

interface ErrorMessageProps {
    message: string
}
export const ErrorMessage: FC <ErrorMessageProps> = ({message}): ReactElement => (
    <div className="flex flex justify-center items-center h-screen">
        <h2 className="font-gotham coral text-5xl text-center pb-4">
            {message}
        </h2>
    </div>
)
