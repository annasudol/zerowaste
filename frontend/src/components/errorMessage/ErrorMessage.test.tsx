import { shallow } from 'enzyme';
import React from 'react';

import { ErrorMessage } from './ErrorMessage';
import toJson from 'enzyme-to-json';

describe('component', (): void => {
  describe('ErrorMessage', (): void => {
    it('match snapshot', (): void => {
      const wrapper = shallow(
        <ErrorMessage message="test"
        />,
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders error message when given prop', (): void => {
      const wrapper = shallow(<ErrorMessage message="test" />);
      expect(wrapper.find('div h2').text()).toBe('test');
    });

    it('renders default error message when prop message was not given', (): void => {
      const wrapper = shallow(<ErrorMessage />);
      expect(wrapper.find('div h2').text()).toBe('Page not found');
    });
  });
});
