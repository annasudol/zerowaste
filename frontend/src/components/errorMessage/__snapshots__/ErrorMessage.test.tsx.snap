// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component ErrorMessage match snapshot 1`] = `
<div
  className="flex flex justify-center items-center h-screen"
>
  <h2
    className="font-gotham coral text-5xl text-center pb-4"
  >
    test
  </h2>
</div>
`;
